server/src/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementServiceImpl.java

```diff
@@ -0,0 +1,532 @@
+                    }
+                    break;
+                case USERNAME:
+                    value = outOfBandManagementConfig.getUsername();
+                    break;
+                case PASSWORD:
+                    value = outOfBandManagementConfig.getPassword();
+                    break;
+            }
+            if (value != null) {
+                optionsBuilder.put(option, value);
+            }
+        }
+        return optionsBuilder.build();
+    }
+
+    private void sendAuthError(final Host host, final String message) {
+        try {
+            hostAlertCache.asMap().putIfAbsent(host.getId(), 0L);
+            Long sentCount = hostAlertCache.asMap().get(host.getId());
+            if (sentCount != null && sentCount <= 0) {
+                boolean concurrentUpdateResult = hostAlertCache.asMap().replace(host.getId(), sentCount, sentCount+1L);
+                if (concurrentUpdateResult) {
+                    final String subject = String.format("Out-of-band management auth-error detected for host:%d in cluster:%d, zone:%d", host.getId(), host.getClusterId(), host.getDataCenterId());
+                    LOG.error(subject + ": " + message);
+                    alertMgr.sendAlert(AlertManager.AlertType.ALERT_TYPE_OOBM_AUTH_ERROR, host.getDataCenterId(), host.getPodId(), subject, message);
+                }
+            }
+        } catch (Exception ignored) {
+        }
+    }
+
+    private boolean transitionPowerState(OutOfBandManagement.PowerState.Event event, OutOfBandManagement outOfBandManagementHost) {
+        if (outOfBandManagementHost == null) {
+            return false;
+        }
+        OutOfBandManagement.PowerState currentPowerState = outOfBandManagementHost.getPowerState();
+        try {
+            OutOfBandManagement.PowerState newPowerState = OutOfBandManagement.PowerState.getStateMachine().getNextState(currentPowerState, event);
+            boolean result = outOfBandManagementDao.updateState(currentPowerState, event, newPowerState, outOfBandManagementHost, null);
+            if (result) {
+                final String message = String.format("Transitioned out-of-band management power state from:%s to:%s due to event:%s for the host id:%d", currentPowerState, newPowerState, event, outOfBandManagementHost.getHostId());
+                LOG.debug(message);
+                ActionEventUtils.onActionEvent(CallContext.current().getCallingUserId(), CallContext.current().getCallingAccountId(), Domain.ROOT_DOMAIN,
+                        EventTypes.EVENT_HOST_OUTOFBAND_MANAGEMENT_POWERSTATE_TRANSITION, message);
+            }
+            return result;
+        } catch (NoTransitionException ignored) {
+            LOG.trace(String.format("Unable to transition out-of-band management power state for host id=%s for the event=%s and current power state=%s", outOfBandManagementHost.getHostId(), event, currentPowerState));
```

Comment:
Consider logging to ``ERROR`` as the system is not behaving as specified and in a completely unexpected manner.  Also, please include the stack trace in the log message to assist with debugging.